### DISCUSSION TITLE
SHA3 Hash Algorithms Adoption

### DISCUSSION THREAD

**Discussion by ChihweiLHBird** [opentofu/opentofu#1449] [Ideas]
Should we adopt SHA3 algorithm to build some new hash functions? I might be able to help to implement, but I am not sure about how to name those functions. For example, we already have SHA2's [`sha256`](https://opentofu.org/docs/language/functions/sha256/), how would we name the SHA3-256 hash function? Maybe like `sha3256`? or `sha3_256`?

**Comment by ghost** [comment]
Hi @ChihweiLHBird HashiCorp has added provider-defined functions to the Terraform provider framework and OpenTofu will support it soon. This would be best implemented in a provider unless there is a strong reason to implement and maintain it in the OpenTofu core.

**Reply by ChihweiLHBird** [reply]
Thank you @janosdebugs!

Since all SHA2 hashing were in the core already (e.g. [SHA2-256](https://opentofu.org/docs/language/functions/filesha256/)), I feel SHA3 can also be maintained in the core and reused cross multiple providers when needed.

Although SHA3 is not yet a popular hashing standard, there are still [some benefits for using it](https://crypto.stackexchange.com/a/95265/555).

We really don't want it in OpenTofu core, we can hold off a little bit here and see if it's needed in the future.
